File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/nullable/KNullableLongPropertyName.java
package com.nagopy.android.kotlinames.property.nullable;

import com.nagopy.android.kotlinames.KNullablePropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KNullableLongPropertyName implements KNullablePropertyName<Long>, KSortablePropertyName<Long> {

    private final String name;

    public KNullableLongPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/nullable/KNullableLongPropertyName.kt
package com.nagopy.android.kotlinames.property.nullable

import com.nagopy.android.kotlinames.KNullablePropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KNullableLongPropertyName(val name: String) : KNullablePropertyName<Long>, KSortablePropertyName<Long> {

    override fun name(): String = name

}
